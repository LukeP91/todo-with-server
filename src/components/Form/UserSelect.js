import { withStyles } from '@material-ui/core/styles'
import FormControl from '@material-ui/core/FormControl'
import InputLabel from '@material-ui/core/InputLabel'
import MenuItem from '@material-ui/core/MenuItem'
import PropTypes from 'prop-types'
import React, { Component } from 'react'
import Select from '@material-ui/core/Select'

const styles = {
  container: {
    marginBottom: '20px',
  },
}

class TextInput extends Component {
  renderInput = props => {
    const {
      field: { name, ...field },
      label,
      classes,
      users,
    } = props

    return (
      <FormControl className={classes.container} id="users-list">
        <InputLabel htmlFor="name-simple">{label}</InputLabel>
        <Select {...field} id="user-select" name={name}>
          {users.map(user => (
            <MenuItem key={user.id} value={user.id}>
              {user.name}
            </MenuItem>
          ))}
        </Select>
      </FormControl>
    )
  }

  render() {
    return this.renderInput(this.props)
  }
}
TextInput.propTypes = {
  classes: PropTypes.object,
  field: PropTypes.object,
  form: PropTypes.object,
  label: PropTypes.string,
  users: PropTypes.array,
}

export default withStyles(styles)(TextInput)
