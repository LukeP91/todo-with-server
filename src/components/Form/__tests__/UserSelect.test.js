import React from 'react'
import { shallow } from 'enzyme'

import UserSelect from '../UserSelect'

describe('UserSelect', () => {
  it('renders correctly', () => {
    const users = [{ id: 1, email: 'test@example.com', name: 'Test user' }]
    const form = { values: { labelsIds: [1] } }
    const field = { name: 'test', value: form.values.labelsIds }

    const wrapper = shallow(<UserSelect field={field} form={form} users={users} />)

    expect(wrapper).toMatchSnapshot()
  })
})
