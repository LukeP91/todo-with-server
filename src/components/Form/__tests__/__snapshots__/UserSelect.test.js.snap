// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserSelect renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithStyles(TextInput)
    field={
      Object {
        "name": "test",
        "value": Array [
          1,
        ],
      }
    }
    form={
      Object {
        "values": Object {
          "labelsIds": Array [
            1,
          ],
        },
      }
    }
    users={
      Array [
        Object {
          "email": "test@example.com",
          "id": 1,
          "name": "Test user",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "classes": Object {
        "container": "TextInput-container-1",
      },
      "field": Object {
        "name": "test",
        "value": Array [
          1,
        ],
      },
      "form": Object {
        "values": Object {
          "labelsIds": Array [
            1,
          ],
        },
      },
      "users": Array [
        Object {
          "email": "test@example.com",
          "id": 1,
          "name": "Test user",
        },
      ],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "classes": Object {
          "container": "TextInput-container-1",
        },
        "field": Object {
          "name": "test",
          "value": Array [
            1,
          ],
        },
        "form": Object {
          "values": Object {
            "labelsIds": Array [
              1,
            ],
          },
        },
        "users": Array [
          Object {
            "email": "test@example.com",
            "id": 1,
            "name": "Test user",
          },
        ],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
